pkg/apis/autoscaling/annotation_validation.go

```diff
@@ -105,6 +110,22 @@ func validateFloats(annotations map[string]string) *apis.FieldError {
 	return errs
 }
 
+func validateScaleDownDelay(annotations map[string]string) *apis.FieldError {
+	var errs *apis.FieldError
+	if w, ok := annotations[ScaleDownDelayAnnotationKey]; ok {
+		if d, err := time.ParseDuration(w); err != nil {
+			errs = apis.ErrInvalidValue(w, ScaleDownDelayAnnotationKey)
+		} else if d < 0 || d > WindowMax {
+			// Since we disallow windows longer than WindowMax, so we should limit this
+			// as well.
+			errs = apis.ErrOutOfBoundsValue(w, 0*time.Second, WindowMax, ScaleDownDelayAnnotationKey)
+		} else if d.Round(time.Second) != d {
+			errs = apis.ErrGeneric("must be specified with at most second precision", ScaleDownDelayAnnotationKey)
+		}
```

Comment:
```suggestion
	switch d, err := time.ParseDuration(w); {
	case err != nil:
		return apis.ErrInvalidValue(w, ScaleDownDelayAnnotationKey)
	case d < 0 || d > WindowMax:
 		// Since we disallow windows longer than WindowMax, so we should limit this
 		// as well.
		return apis.ErrOutOfBoundsValue(w, 0*time.Second, WindowMax, ScaleDownDelayAnnotationKey)		
	case d.Round(time.Second) != d:
		return apis.ErrGeneric("must be specified with at most second precision", ScaleDownDelayAnnotationKey)
	}
```